manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml

```diff
@@ -21,12 +25,12 @@ metadata:
 spec:
   selector:
     matchLabels:
-      istio.io/gateway-name: {{.Name}}
+      gateway.networking.k8s.io/gateway-name: {{.Name}}
```

Comment:
big oof. This field is immutable...

Reply:
Oh crap...I honestly don't know if there's a way around this without an imperative one-time migration (yuck)

Reply:
The deployment controller doesn't see the actual deployments/services though; it only sees the `Gateway`. So it has no way of knowing whether a given `Gateway` using the old label or the new one

Reply:
It can, we already do similar actually. See `canManage`. Or just `d.deployments`.

All of this is bad controller practices, though... but it is what it is...

Reply:
Oh wow I wouldn't expect us to have this info; if we're already doing it...sure I guess? I'll pick this back up then

Reply:
Got it working. This is ready for review; just dealing with some linting/testing errors